R/fit_baselines_one_location.R

```diff
@@ -31,6 +31,22 @@
 #'   - window_size (integer), determines how many previous observations inform
 #'     the forecast
 #'
+#' Additionally, this function will return slightly different output forecasts
+#' depending on the relationship between the `reference_date`, requested `horizons`,
+#' and dates contained within `target_ts`. There are three possible cases:
+#'   1. The requested forecasts begin exactly one time unit (given by the
+#'      temporal_resolution) after the last observed value. Here, no changes are
+#'      made to the returned forecasts.
+#'   2. The requested forecasts begin two or more time units after the last
+#'      observed value. Here, we make predictions starting from the next time unit
+#'      after the last observed value until the last requested date, returning
+#'      only forecasts for the requested dates.
+#'   3. The dates for the requested forecasts overlap partially or completely with
+#'      observed values contained within `target_ts`. Here, any forecasted values
+#'      for overlapping dates are replaced by the associated observed values.
+#'
+#' Note that we warn for the second and third cases.
```

Comment:
That's good point — I had been thinking mostly of making weekly flu forecasts which won't trigger any of the warnings, but the daily predictions we make that inform the trends ensemble will trigger a warning for the second case. I may still keep the warning for the third case, as it may be a useful reminder if we ever retrospectively make forecasts